extras/docker/fromsource/heketi-start.sh

```diff
@@ -6,75 +6,83 @@
 
-echo "Setting up heketi database"
+info() {
+    echo "$*" | tee -a "$LOG"
+}
 
-if [[ -d "${HEKETI_PATH}" ]]; then
-    # Test that our volume is writable.
-    touch "${HEKETI_PATH}/test" && rm "${HEKETI_PATH}/test"
-    if [ $? -ne 0 ]; then
-        echo "${HEKETI_PATH} is read-only"
-        exit 1
-    fi
+error() {
+    echo "error: $*" | tee -a "$LOG" >&2
+}
 
-    if [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; then
-        echo "No database file found" | tee -a "${HEKETI_PATH}/container.log"
-        out=$(mount | grep "${HEKETI_PATH}" | grep heketidbstorage)
-        if [[ $? -eq 0 ]]; then
-            echo "Database volume found: ${out}" | tee -a "${HEKETI_PATH}/container.log"
-            echo "Database file is expected, waiting..." | tee -a "${HEKETI_PATH}/container.log"
-            check=0
-            while [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; do
-                sleep 5
-                if [[ ${check} -eq 5 ]]; then
-                   echo "Database file did not appear, exiting." | tee -a "${HEKETI_PATH}/container.log"
-                   exit 1
-                fi
-                ((check+=1))
-            done
-        fi
-    fi
+info "Setting up heketi database"
 
-    stat "${HEKETI_PATH}/heketi.db" | tee -a "${HEKETI_PATH}/container.log"
-    # Workaround for scenario where a lock on the heketi.db has not been
-    # released.
-    # This code uses a non-blocking flock in a loop rather than a blocking
-    # lock with timeout due to issues with current gluster and flock
-    # ( see rhbz#1613260 )
-    for _ in $(seq 1 60); do
-        flock --nonblock "${HEKETI_PATH}/heketi.db" true
-        flock_status=$?
-        if [[ $flock_status -eq 0 ]]; then
-            break
-        fi
-        sleep 1
-    done
-    if [[ $flock_status -ne 0 ]]; then
-        echo "Database file is read-only" | tee -a "${HEKETI_PATH}/container.log"
+# Ensure the data dir exists
+mkdir -p "${HEKETI_PATH}" 2>/dev/null
+if [[ $? -ne 0 && ! -d "${HEKETI_PATH}" ]]; then
+    error "Failed to create ${HEKETI_PATH}"
+    exit 1
+fi
+
+# Test that our volume is writable.
+touch "${HEKETI_PATH}/test" && rm "${HEKETI_PATH}/test"
+if [ $? -ne 0 ]; then
+    error "${HEKETI_PATH} is read-only"
+    exit 1
+fi
+
+if [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; then
+    info "No database file found"
+    out=$(mount | grep "${HEKETI_PATH}" | grep heketidbstorage)
+    if [[ $? -eq 0 ]]; then
+        info "Database volume found: ${out}"
+        info "Database file is expected, waiting..."
+        check=0
+        while [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; do
+            sleep 5
+            if [[ ${check} -eq 5 ]]; then
+               error "Database file did not appear, exiting."
+               exit 1
+            fi
+            ((check+=1))
+        done
     fi
-else
-    mkdir -p "${HEKETI_PATH}"
-    if [[ $? -ne 0 ]]; then
-        echo "Failed to create ${HEKETI_PATH}"
-        exit 1
+fi
+
+stat "${HEKETI_PATH}/heketi.db" | tee -a "${LOG}"
```

Comment:
Can we do something like this?
```
info "$(stat "${HEKETI_PATH}/heketi.db")"
```

Reply:
I want to re-do this logic a bit but I figured I would do that in a different pr, in my mind, this one was just for cleaning up the existing code to make it more readable/convenient w/o much changes to the logic or behavior.

Reply:
Well, this would not change anything really, it would result in the stat output being put into the logs, as before, right? that's fine, we can do it later...